src/twisted/names/dns.py

```diff
@@ -378,7 +379,37 @@ def _isSubdomainOf(descendantName, ancestorName):
     return descendantLabels[-len(ancestorLabels) :] == ancestorLabels
 
 
-def str2time(s):
+# helper function for str2time, needed to appease mypy see GH-11956
+def _str2time(s: str) -> int:
```

Comment:
Thanks for the comment.
It still don't make sense ... according to my poor brain.

Instead of `gh-1245` I think is best add the full url.
I wasn't sure if this a mypy bug reference of twisted bug reference.

maybe have something  like this.

```suggestion
def _str2time(s: str) -> int:
    """
    mypy doesn't like type-punning str | bytes | int | None into a str so we have this helper function.
    """
```